### ISSUE TITLE
False positive pattern in loop pointer rule

### ISSUE THREAD

**Issue by romdr** [returntocorp/sgrep-rules#2972]
This [pattern](https://github.com/returntocorp/semgrep-rules/blob/79360be4bb8ad1ececceb50f7cdc84f3086681cb/go/lang/correctness/looppointer.yaml#L28) seems incorrect:

```
      - pattern: |
          for _, $VALUE := range $SOURCE {
            <... $ANYTHING(..., <... $VALUE ...>, ...) ...>
          }
```

This triggers a semgrep warning:

```
	input := []string{"a", "b", "c"}
	output := []string{}

	for _, val := range input {
		output = append(output, val)
	}
```

There is no pointer involved here.

Go Playground example [here](https://go.dev/play/p/mkOmtUBxQ_T).

Semgrep Playground example:
![image](https://github.com/returntocorp/semgrep-rules/assets/2272368/0c7e725e-8888-4e1c-9b72-a5f3ae7d6c28)

**Comment by enncoded** [comment]
Hey! Thanks for filing this issue :)

I don't think Semgrep can do that kind of type analysis over contents of a collection in Go right now. I filed an issue to the engine team https://github.com/returntocorp/semgrep/issues/8117, so hopefully this should be fixed soon!
